.github/workflows/main.yml

```diff
@@ -24,4 +24,13 @@ jobs:
 
       - name: Run pytest
         run: pytest devolearn/tests/test.py
-     
\ No newline at end of file
+
+      - name: Generate Report
+        run: |
+          pip install coverage
+          coverage run -m pytest devolearn/tests/test.py
```

Comment:
It seems like you're running the tests twice. Which just takes up double the time. I suggest you to [follow this example](https://github.com/Mayukhdeb/torch-dreams/blob/7afe15fc7ee5540ae7f03a0f3d54c721a7e1b5c5/.github/workflows/main.yml#L34) if possible.